independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java

```diff
@@ -303,13 +321,23 @@ void notify(T event, ObserverExceptionHandler exceptionHandler, boolean async) {
                                 // See for instance discussions on https://github.com/eclipse-ee4j/cdi/issues/467
                                 txManager.getTransaction().registerSynchronization(sync);
                                 // registration succeeded, notify all non-tx observers synchronously
-                                predicate = predicate.and(this::isNotTxObserver);
+                                predicate = predicate.and(new Predicate<>() {
+                                    @Override
+                                    public boolean test(ObserverMethod<? super T> observerMethod) {
+                                        return isNotTxObserver(observerMethod);
+                                    }
+                                });
```

Comment:
Don't you generate a bit more allocations this way. I would expect `notify()` to be called quite often?

Reply:
The reason I drop the method reference is that it brings some overhead at startup. I doubt the allocations make much of a difference with this change, but I could always check the flamegraph

Reply:
Yeah, I wouldn't have mentioned it but I wonder if it might not be a hot path if you have a lot of events.

@mkouba would know better.